aten/src/ATen/native/cuda/Activation.cu

```diff
@@ -2,9 +2,239 @@
 #include "ATen/NativeFunctions.h"
 #include "ATen/Dispatch.h"
 #include "ATen/cuda/CUDAApplyUtils.cuh"
+#include "ATen/cuda/detail/IndexUtils.cuh"
+
 
 namespace at { namespace native {
 
+// -----------------------------------
+// prelu forward
+// -----------------------------------
+template <typename scalar_t>
+void prelu_cuda_kernel_share_weights(
+  const Tensor& input,
+  Tensor& result,
+  const scalar_t* weight_data) {
+
+  at::cuda::CUDA_tensor_apply2<scalar_t, scalar_t>(
+    input,
+    result,
+    [=] __device__ (
```

Comment:
Not sure about the perf difference. But passing by reference might be faster consider that a pointer is pretty wide on GPU. You also try porting the exact same apply kernel over (just temporarily) and see if the problem is the apply fn.

Reply:
just realize devgpu ash and prn have different GPUs (K40m vs M40)... I was running the perf experiments on these two types of machines...I guess that's why there is a difference..

Reply:
haha I see :)